op-chain-ops/deployer/pipeline/implementations.go

```diff
@@ -0,0 +1,101 @@
+package pipeline
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"os"
+
+	"github.com/ethereum-optimism/optimism/op-chain-ops/deployer/opsm"
+	"github.com/ethereum-optimism/optimism/op-chain-ops/deployer/state"
+	"github.com/ethereum-optimism/optimism/op-chain-ops/foundry"
+	"github.com/ethereum-optimism/optimism/op-chain-ops/script"
+)
+
+func DeployImplementations(ctx context.Context, env *Env, intent *state.Intent, st *state.State) error {
+	lgr := env.Logger.New("stage", "deploy-implementations")
+
+	if !shouldDeployImplementations(intent, st) {
+		lgr.Info("implementations deployment not needed")
+		return nil
+	}
+
+	lgr.Info("deploying implementations")
+
+	var artifactsFS foundry.StatDirFs
+	var err error
+	if intent.ContractArtifactsURL.Scheme == "file" {
+		fs := os.DirFS(intent.ContractArtifactsURL.Path)
+		artifactsFS = fs.(foundry.StatDirFs)
+	} else {
+		return fmt.Errorf("only file:// artifacts URLs are supported")
+	}
+
+	var dump *foundry.ForgeAllocs
+	var dio opsm.DeployImplementationsOutput
+	err = CallScriptBroadcast(
+		ctx,
+		CallScriptBroadcastOpts{
+			L1ChainID:   big.NewInt(int64(intent.L1ChainID)),
+			Logger:      lgr,
+			ArtifactsFS: artifactsFS,
+			Deployer:    env.Deployer,
+			Signer:      env.Signer,
+			Client:      env.L1Client,
+			Handler: func(host *script.Host) error {
+				host.SetEnvVar("IMPL_SALT", st.Create2Salt.Hex()[2:])
+				host.ImportState(st.SuperchainDeployment.StateDump)
+				dio, err = opsm.DeployImplementations(
+					host,
+					opsm.DeployImplementationsInput{
+						WithdrawalDelaySeconds:          big.NewInt(604800),
+						MinProposalSizeBytes:            big.NewInt(126000),
+						ChallengePeriodSeconds:          big.NewInt(86400),
+						ProofMaturityDelaySeconds:       big.NewInt(604800),
+						DisputeGameFinalityDelaySeconds: big.NewInt(302400),
+						Release:                         "op-contracts/v1.6.0",
```

Comment:
Fine to keep this here for now, but the op-contracts version and constants should probably be managed somewhere where they can be tested / made consistent with a source of truth.

Reply:
Yep agreed. I have a milestone for that on the tracker sheet.